zk/stages/stage_sequence_execute_transactions.go

```diff
@@ -26,75 +26,55 @@ import (
 	"github.com/ledgerwatch/log/v3"
 )
 
-func getNextPoolTransactions(cfg SequenceBlockCfg, executionAt, forkId uint64, alreadyYielded mapset.Set[[32]byte]) ([]types.Transaction, error) {
+func getNextPoolTransactions(ctx context.Context, cfg SequenceBlockCfg, executionAt, forkId uint64, alreadyYielded mapset.Set[[32]byte]) ([]types.Transaction, error) {
 	var transactions []types.Transaction
 	var err error
 	var count int
-	killer := time.NewTicker(50 * time.Millisecond)
-LOOP:
-	for {
-		// ensure we don't spin forever looking for transactions, attempt for a while then exit up to the caller
-		select {
-		case <-killer.C:
-			break LOOP
-		default:
+	// killer := time.NewTicker(300 * time.Millisecond)
+	gasLimit := utils.GetBlockGasLimitForFork(forkId)
+
+	if err := cfg.txPoolDb.View(ctx, func(poolTx kv.Tx) error {
+		slots := types2.TxsRlp{}
+		_, count, err = cfg.txPool.YieldBest(preForkId11TxLimit, &slots, poolTx, executionAt, gasLimit, alreadyYielded)
+		if err != nil {
+			return err
 		}
-		if err := cfg.txPoolDb.View(context.Background(), func(poolTx kv.Tx) error {
-			slots := types2.TxsRlp{}
-			_, count, err = cfg.txPool.YieldBest(yieldSize, &slots, poolTx, executionAt, utils.GetBlockGasLimitForFork(forkId), alreadyYielded)
-			if err != nil {
-				return err
-			}
-			if count == 0 {
-				time.Sleep(500 * time.Microsecond)
-				return nil
-			}
-			transactions, err = extractTransactionsFromSlot(&slots)
-			if err != nil {
-				return err
-			}
+		if count == 0 {
```

Comment:
Do we need the sleep here as the caller will sleep 250ms if nothing is returned from this function?